### ISSUE TITLE
Prepending current PATH to PATH defined in "terminal.integrated.env.osx"

### ISSUE THREAD

**Issue by rongduan-zhu** [Microsoft/vscode#34337]
- VSCode Version: 1.15.1
- OS Version: macOS Sierra 10.12.6

Steps to Reproduce:

1. Update workspace settings to something like
```json
{
  "terminal.integrated.env.osx": {
    "PATH": "/home"
  }
}
```
2. Now open Terminal and look at PATH
```bash
echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/home
```

On Windows, having the same setting will not result in the current `PATH` being prepended, but instead, just setting the path to whatever is defined in `terminal.integrated.env.*`

Reproduces without extensions: Yes

**Comment by Tyriar** [comment]
Unless there's a more intuitive way of doing this through the `terminal.integrated.env.*` settings, I'd rather not support it. This would make replacing PATH impossible and inconsistent with how other keys work.

There is a workaround though by using something like [`--rcfile`](https://unix.stackexchange.com/a/152735/115410) to run a script on init.

**Comment by rongduan-zhu** [comment]
Hi @Tyriar , my description was confusing. 
So on Mac, with the setting below
```json
{
  "terminal.integrated.env.osx": {
    "PATH": "/home"
  }
}
```
the resulting PATH in terminal in VS Code will be something like `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/home`. `/usr/local/bin:/usr/bin:/bin:/usr/sbin` was added to `terminal.integrated.env.osx.PATH`

Whereas on Windows, having the "same" setting
```json
{
  "terminal.integrated.env.windows": {
    "PATH": "C:\bin"
  }
}
```
the resulting PATH in the integrated terminal will be just `C:\bin`, nothing else will be added to PATH besides whats defined for `terminal.integrated.env.windows.PATH`.

Having different behavior for Windows and Mac is strange. Also, on Mac, it seems that only the `PATH` key in `terminal.integrated.env.osx` is being changed, we have other env variables defined in `terminal.integrated.env.osx` and they all stayed the same, this is super confusing...

The reason I think this is an issue is that our app allows the user to open a project in our app using VS Code, and our app has a custom Python installed. We use `terminal.integrated.env.osx.PATH` to essentially tell the terminal to use the Python installed by our app instead of the Python in, say, `/usr/local/bin`. This works fine on Windows, but on Mac, things are broken because VS Code is modifying `terminal.integrated.env.osx.PATH`.

**Comment by Tyriar** [comment]
It's the same behavior, it's just that Windows PATH is typically set and complete when you login into Windows whereas on Mac it has more entries added to it through the `~/.bash_profile` and `~/.bashrc` files.

> Also, on Mac, it seems that only the PATH key in terminal.integrated.env.osx is being changed, we have other env variables defined in terminal.integrated.env.osx and they all stayed the same, this is super confusing...

Are you sure your start up scripts are not overriding the values here?

> but on Mac, things are broken because VS Code is modifying terminal.integrated.env.osx.PATH

It's fairly typical to set the system's python version using alias in one of these files. Could that be messing with it?

**Comment by rongduan-zhu** [comment]
Hi @Tyriar , thank you for providing pointers on where to check. I've checked the Mac I'm using, there is no `~/.bash_profile` or `~/.bashrc` file. After some more investigation, looks like on Mac, there is a `path_helper` function which appends paths defined in `/etc/paths` and `/etc/paths.d` to `PATH`. And that is invoked in `/etc/profile`. I think this is causing the issue.

Thank you for the help, agree with you that this is not a VS Code issue.

**Comment by Tyriar** [comment]
Apparently tasks.json supports the syntax `${env:Name}`. So I'll reopen this feature request to allow the following:

```js
{
  "terminal.integrated.env.windows": {
    "PATH": "${env:PATH};C:\bin"
  }
}
```

@dbaeumer does this sound consistent with what's supported in tasks.json?

**Comment by dbaeumer** [comment]
@Tyriar yes, `${env:Name}` is consistent with tasks and debug. There is a configuration resolver service you can use to get the values.

**Comment by Tyriar** [comment]
@dbaeumer great!

This is open to PRs 😃

**Comment by pvgoran** [comment]
This would be a useful feature to have. Just now, I was searching for a way to amend the `PATH` environment variable, and I was quite surprised that there is no straightforward way to do it. (For now, I went with specifying `--rcfile` in `terminal.integrated.shellArgs.linux`, but it's far from a clean solution.)

**Comment by JohnKozell** [comment]
+1, just tried appending or prepending the path in user settings, does not work.
Moreover, the terminal environment variables set in user settings, do not seem to apply to tasks - am I understanding that correct?
Another way around this, might be to enable tasks to run multiple commands, then I could add a batch file to set environment variables, such as Path.

**Comment by Tyriar** [comment]
Related: https://github.com/Microsoft/vscode/issues/2809
